lib/facter/conjur.rb

```diff
@@ -21,9 +21,9 @@ def credentials netrc_path, url
           when 'machine'
             found = value.start_with?(uri.to_s) || value == uri.host
           when 'login'
-            login = value
+            login = value if found
           when 'password'
-            password = value
+            password = value if found
           end
           return [login, password] if login && password
```

Comment:
This method is going to return `nil` if it doesn't find an entry for `uri`. The caller, below, doesn't check for a `nil` return value, it just assumes that it can call `first` on the result. So, in the case where it doesn't find the `uri`, it'll log a `NoSuchMethod` exception.

It seems like the caller should check for `nil` and do something reasonable: ignore it if a missing entry is expected, log a better message if it's not expected.